src/SFML/Graphics/Sprite.cpp

```diff
@@ -131,36 +122,35 @@ void Sprite::draw(RenderTarget& target, RenderStates states) const
     states.transform *= getTransform();
     states.texture        = m_texture;
     states.coordinateType = CoordinateType::Pixels;
-    target.draw(m_vertices.data(), m_vertices.size(), PrimitiveType::TriangleStrip, states);
-}
-
-
-////////////////////////////////////////////////////////////
-void Sprite::updatePositions()
-{
-    const FloatRect bounds = getLocalBounds();
 
-    m_vertices[0].position = Vector2f(0, 0);
-    m_vertices[1].position = Vector2f(0, bounds.height);
-    m_vertices[2].position = Vector2f(bounds.width, 0);
-    m_vertices[3].position = Vector2f(bounds.width, bounds.height);
+    target.draw(m_vertices.data(), m_vertices.size(), PrimitiveType::TriangleStrip, states);
 }
 
 
 ////////////////////////////////////////////////////////////
-void Sprite::updateTexCoords()
+void Sprite::updateVertices()
 {
-    const FloatRect convertedTextureRect(m_textureRect);
-
-    const float left   = convertedTextureRect.left;
-    const float right  = left + convertedTextureRect.width;
-    const float top    = convertedTextureRect.top;
-    const float bottom = top + convertedTextureRect.height;
-
-    m_vertices[0].texCoords = Vector2f(left, top);
-    m_vertices[1].texCoords = Vector2f(left, bottom);
-    m_vertices[2].texCoords = Vector2f(right, top);
-    m_vertices[3].texCoords = Vector2f(right, bottom);
+    assert(m_textureRect.width >= 0);
+    assert(m_textureRect.height >= 0);
```

Comment:
I don't think this can ever be negative unless a user explicitly passes a negative texture rect in `setTextureRect`, because the constructor from `sf::Texture` converts the texture's size which is unsigned.

Reply:
Which begs the question whether we should "allow" negative texture rects in the first place.

Since we don't allow it here, I think it would be better to just move the asserts into `setTextureRect` and adjust the documentation.

Reply:
`setTextureRect` is not the only place where `m_textureRect` is set, there's also the constructor -- I decided to put the assertion here because both the constructor and `setTextureRect` end up calling `updateVertices`.

I will adjust the documentation to clarify that negative rects are not valid. I think the right solution here would be to accept a `UIntRect` instead of `IntRect`, but I'd prefer leaving that to a separate PR as it's a more breaking change than the ones presented here.

Reply:
Negative texture rect size can be used to flip the represented portion of the texture horizontally and/or vertically. Why disallow this? It looks like a valid use case to me.

Reply:
@kimci86: is this well-defined and/or documented anywhere in SFML? I searched a bit in the codebase but couldn't find anything. If it's something we want to support, we could document it in `setTextureRect` and the sprite ctor. Thoughts? @eXpl0it3r

Reply:
I thought that was a well-known behavior but now I don't see documentation or tutorial about it. I guess I saw the implementation a while ago and learned from that.

Reply:
> I thought that was a well-known behavior but now I don't see documentation or tutorial about it. I guess I saw the implementation a while ago and learned from that.

Do you know if this works on every platform and is compatible with GLES2 and our eventual plans of moving to a more modern GL implementation?

I've briefly looked around and didn't find much information, I did see some people saying that GLES2 on Android might not support UVs outside [0..1] range, but that wasn't very conclusive either.

Reply:
I don't really see a reason to restrict it. I can in theory do the same with texture coordinates when using a vertex array, so why artificially restrict it here. Especially in light that it was previously not constraint.

If we want, we could potentially assert that `size - pos` doesn't land in a negative quadrant, which is the constraint I was thinking about, while not considering that position can of course be non-zero.

I don't think it's necessary to know what GLES2 will or will not support. It's and edge case limitation we can introduce at the time when it becomes a constraint.